opendatahub-io/ilab-on-ocp
# Parameterize TLS configuration options

Update Pipeline to allow TLS Cert verification when connecting to Teacher- and Judge- models

## Description
- Add `_use_tls` switch params for sdg, mt_bench, and final_eval phases
- Source certs from teacher- and judge-server ConfigMaps (already used for other metadata)
- Update eval phases to create and use httpx client instead of a custom-patched openai client
  - (Required update after [update to ilab eval](https://github.com/instructlab/eval/commit/010ff8c11db4e2d3b4812ec8bdf6b6ef94442687#diff-e5e89be967cd71d6e6a091b6f72126552782a6da4abac9c0e0b9c9fac33b2745L124)
- Update README 

## How Has This Been Tested?
- Run pipeline with certs and _use_tls params enabled, verify successful connections to both teacher and judge models during SDG and Eval phases
- Run pipeline with certs turned off (and models requiring them), verify phases fail due to incorrect/missing certs error

## Merge criteria:
